docs/modules/ROOT/pages/adr/0045-service-orchestration-crossplane-2-0.adoc

```diff
@@ -0,0 +1,136 @@
+:adr_date:
+:adr_upd_date:
+:adr_status: draft
+:adr_tags: framework,framework2,crossplane,composition,orchestration
+
+include::partial$adr-meta.adoc[]
+
+[NOTE]
+.Summary
+====
+This ADR defines the service orchestration mechanism for AppCat Framework 2.0, evaluating Crossplane 2.0 composition functions against a custom Kubernetes operator.
+
+The decision selects https://www.crossplane.io/[Crossplane] 2.0 with composition functions as the orchestration layer, providing generic service management with built-in revision control and clear separation between framework code (https://go.dev/[Go]) and service configuration (https://www.kcl-lang.io/[KCL]).
+====
+
+== Context
+
+=== Problem Statement
+
+The AppCat Framework 2.0 requires a service orchestration mechanism that:
+
+* Renders Helm charts based on service configuration and user-selected parameters
+* Manages complete service lifecycle (create, update, delete)
+* Handles connection secrets and billing/maintenance/backup/metrics integration
+* Allows Framework Engineers to maintain core logic in Go
+* Enables Service Maintainers to add new services without touching Go code
+* Provides versioning and revision management for controlled rollouts
+
+=== Architectural Context
+
+The framework architecture separates concerns across three layers:
+
+1. **Framework Engineers** maintain the orchestration runtime (Go composition functions)
+2. **Service Maintainers** define service configurations (KCL compiled managed resource objects)
+3. **Service Users** consume services via declarative API (XRD/Composite resources)
+
+The orchestration layer sits at the boundary between framework runtime and service configuration, responsible for:
+
+* Loading service configurations at runtime
+* Merging static defaults with dynamic user parameters
+* Generating Kubernetes resources (HelmRelease, Schedule, ServiceMonitor, secrets, etc.)
+* Tracking revisions for upgrade workflows
+* Propagating status and errors to users
+
+== Options
+
+=== Option A: Crossplane 2.0 with Composition Functions
+
+==== Description
+
+https://www.crossplane.io/[Crossplane] 2.0 composition functions execute in pipeline mode during reconciliation, receiving composite state and returning desired resources.
+A single generic function handles all services by loading service-specific configuration Kubernetes like object (ex:, `RedisServiceConfig`) at runtime, merging user parameters, and generating resources.
+Other functions can be added to increase framework quality such as increased service resilience and error handling.
+
+==== Advantages
+
+* **Generic Function Pattern**: Single function handles all services without Go code changes
+* **Built-in Composition Revisions**: Native versioning and gradual rollout capabilities
+* **Clear Separation**: Framework code (Go) completely separated from service configuration (KCL)
+* **Provider Ecosystem**: Leverage existing Crossplane providers (Helm, Kubernetes) instead of custom implementations
```

Comment:
```suggestion
* **Provider Ecosystem**: Leverage existing Crossplane providers (Helm, SQL) instead of custom implementations
```
I wouldn't put `provider-kubernetes` as an example if we don't plan to use it :-)